intelmq/bots/parsers/shadowserver/config.py

```diff
@@ -1970,3 +1973,113 @@ def convert_date(value):
         'classification.type': 'other',
     },
 }
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Amplification-DDoS-Victim
+amplification_ddos_victim = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+    ],
+    'optional_fields': [
+        ('source.port', 'src_port'),
+        ('protocol.transport', 'protocol'),
+        ('destination.port', 'dst_port'),
+        ('source.reverse_dns', 'hostname'),
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('extra.', 'tag', validate_to_none),
+        ('extra.', 'naics', invalidate_zero),
+        ('extra.', 'sic', invalidate_zero),
+        ('extra.', 'request', validate_to_none),
+        ('extra.', 'count', convert_int),
+        ('extra.', 'bytes', convert_int),
+        ('extra.', 'sensor_geo', validate_to_none),
+        ('extra.', 'sector', validate_to_none),
+        ('extra.', 'end_time', validate_to_none),
+        ('extra.', 'public_source', validate_to_none),
+    ],
+    'constant_fields': {
+        'classification.taxonomy': 'availability',
+        'classification.type': 'ddos',
+        'classification.identifier': 'amplification-ddos-victim',
+    }
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/HTTP-Scanners
+http_scanners = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+        ('source.port', 'port'),
+    ],
+    'optional_fields': [
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('source.reverse_dns', 'hostname'),
+        ('destination.ip', 'dst_ip'),
+        ('destination.port', 'dst_port'),
+        ('destination.asn', 'dst_asn'),
+        ('destination.geolocation.cc', 'dst_geo'),
+        ('destination.fqdn', 'dst_dns', validate_fqdn),
+        ('extra.', 'type', validate_to_none),
+        ('extra.', 'naics', invalidate_zero),
+        ('extra.', 'sic', invalidate_zero),
+        ('extra.', 'sector', validate_to_none),
+        ('extra.destination.sector', 'dst_sector', validate_to_none),
+        ('extra.', 'public_source', validate_to_none),
+        ('extra.', 'sensorid', validate_to_none),
+        ('extra.', 'pattern', validate_to_none),
+        ('extra.', 'url', validate_to_none),
+        ('extra.file.md5', 'file_md5', validate_to_none),
+        ('extra.file.sha256', 'file_sha256', validate_to_none),
+        ('extra.', 'request_raw', validate_to_none),
+    ],
+    'constant_fields': {
+        'classification.taxonomy': 'information gathering',
+        'classification.type': 'scanner',
+        'classification.identifier': 'http-scanners',
```

Comment:
I'd only use the protocol `http` here. It is already clear from the type that this is a scanner. That would be also in line with parser dealing with similar data. And we can also add `protocol.application` and `protocol.transport`

Reply:
I agree 'http' should be sufficient for the identifier.

Reply:
Btw, in the Reference Security Incident Classification Taxonomy the corresponding type is "scanning". So maybe we should change "scanner" to "scanning" for the data harmonization?

Reply:
protocol.application should be set to 'http' and protocol.transport to 'tcp' for all events from this feed.

Reply:
> Btw, in the Reference Security Incident Classification Taxonomy the corresponding type is "scanning". So maybe we should change "scanner" to "scanning" for the data harmonization?

No, it's scanner: https://github.com/enisaeu/Reference-Security-Incident-Taxonomy-Task-Force/blob/master/working_copy/machinev1#L53
The `value` is the machine-readable specifier, expanded is for humans.